segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/DefragmentOperation.java

```diff
@@ -269,9 +264,8 @@
                                 bytesToRead -= size;
                                 readAtOffset += size;
                                 return chunkedSegmentStorage.getChunkStorage().write(writeHandle, writeAtOffset, size, new ByteArrayInputStream(buffer, 0, size))
-                                        .thenApplyAsync(written -> {
+                                        .thenAcceptAsync(written -> {
                                             writeAtOffset += written;
```

Comment:
This is not thread safe. Use `AtomicLong`.
While `volatile` is atomic with respect to get and set, it is not with respect to modification. Your addition is a read-modify-update operation which is not atomic.


Please fix this everywhere in your code.

Reply:
Based to application logic they are thread safe as only one lambda from this method executes at a time.  However to avoid future bugs and confusion I'll fix where values are modified. (based on code warning in IDE)

Reply:
ok updated.